sklearn/metrics/_regression.py

```diff
@@ -395,21 +395,30 @@ def mean_absolute_percentage_error(
     >>> mean_absolute_percentage_error(y_true, y_pred)
     112589990684262.48
     """
+    input_arrays = [y_true, y_pred, sample_weight, multioutput]
+    xp, _ = get_namespace(*input_arrays)
+    dtype = _find_matching_floating_dtype(y_true, y_pred, sample_weight, xp=xp)
+
     y_type, y_true, y_pred, multioutput = _check_reg_targets(
         y_true, y_pred, multioutput
     )
     check_consistent_length(y_true, y_pred, sample_weight)
-    epsilon = np.finfo(np.float64).eps
-    mape = np.abs(y_pred - y_true) / np.maximum(np.abs(y_true), epsilon)
-    output_errors = np.average(mape, weights=sample_weight, axis=0)
+    epsilon = xp.asarray(xp.finfo(xp.float64).eps, dtype=dtype)
+    y_true_abs = xp.asarray(xp.abs(y_true), dtype=dtype)
+    mape = xp.asarray(xp.abs(y_pred - y_true), dtype=dtype) / xp.where(
+        epsilon < y_true_abs, y_true_abs, epsilon
+    )
```

Comment:
I think we can also use xp.maximum directly.
```suggestion
    mape = xp.asarray(xp.abs(y_pred - y_true), dtype=dtype) / xp.maximum(
        y_true_abs, epsilon
    )
```

Reply:
Thank you @OmarManzoor. I believe `xp.maximum` is not supported in the `array_api_strict` version that we are using (v2022.12). It exists starting from v2023.12. 
[PR for maximum and minimum spec](https://github.com/data-apis/array-api/pull/713)

Reply:
We are supporting the latest version now.
https://github.com/scikit-learn/scikit-learn/blob/b0f86e797190d1fd62331eac8b88a5b0dfa522b9/sklearn/utils/_array_api.py#L572

Reply:
Oh awesome, thank you! The merge somehow didn't capture this change on my local repo, let me try again.

Reply:
Right now on my local I can see the 2023.12 in the code, but on pytest I am still getting`AttributeError: module 'array_api_strict' has no attribute 'maximum'`. I think this means I need to update my `array_api_strict` but I can't seem to find the correct command to update that... could you please advise? thank you

Reply:
Can you please update array-api-strict to the latest version on your system? I think that should resolve this.
```
pip install array-api-strict==2.0.1
```

Reply:
Done, thank you!